components/analytics-core/org.wso2.carbon.analytics.datasource.core.test/src/main/java/org/wso2/carbon/analytics/datasource/core/AnalyticsDataServiceTest.java

```diff
@@ -649,47 +680,47 @@ public void testRecordAddRetrieveWithKeyValues() throws AnalyticsException {
         Assert.assertEquals(new HashSet<>(recordsIn), matchRecords);
         this.cleanupTable(tenantId, tableName);
     }
-    
-    @Test (dependsOnMethods = "testRecordAddRetrieveWithKeyValues")
+
+    @Test(dependsOnMethods = "testRecordAddRetrieveWithKeyValues")
     public void testDataRecordAddReadPerformanceNonIndex() throws AnalyticsException {
         int tenantId = 51;
         String tableName = "TableX";
         this.cleanupTable(tenantId, tableName);
-        System.out.println("\n************** START ANALYTICS DS (WITHOUT INDEXING) PERF TEST **************");
+        logger.info("\n************** START ANALYTICS DS (WITHOUT INDEXING) PERF TEST **************");
         int n = 100, batch = 200;
         List<Record> records;
-        
+
         /* warm-up */
-        this.service.createTable(tenantId, tableName);      
+        this.service.createTable(tenantId, tableName);
         for (int i = 0; i < 10; i++) {
             records = AnalyticsRecordStoreTest.generateRecords(tenantId, tableName, i, batch, -1, -1);
             this.service.put(records);
         }
         this.cleanupTable(tenantId, tableName);
-        
+
         this.service.createTable(tenantId, tableName);
         long start = System.currentTimeMillis();
         for (int i = 0; i < n; i++) {
             records = AnalyticsRecordStoreTest.generateRecords(tenantId, tableName, i, batch, -1, -1);
             this.service.put(records);
         }
         long end = System.currentTimeMillis();
-        System.out.println("* Records: " + (n * batch));
-        System.out.println("* Write Time: " + (end - start) + " ms.");
-        System.out.println("* Write Throughput (TPS): " + (n * batch) / (double) (end - start) * 1000.0);
+        logger.info("* Records: " + (n * batch));
```

Comment:
This is not a performance test. hence, I don't see a point of recording number of processed records, execution time, throughput or any other time measurement.  Even the future changes causes to increase the execution time or reduce the throughput, this test case doesn't fail. Hence, recording them is useless.

Noticed similar logs in other places. Please remove them all.

Reply:
This is a perf test done locally. simply printing the time taken to persist the given number of records. This test helps us compare the throughput of different recordstores in a local environment. Also This test runs after all the unit tests written for records stores. I dont see any harm running a simple perf test locally after running the "actual" unit tests for a recordstore

Reply:
> This is a perf test done locally. simply printing the time taken to persist the given number of records. This test helps us compare the throughput of different recordstores in a local environment.

Can you explain why would you want to run a pref-test locally? Whose going to read these logs and compare with previous ones in the build process? :)

Correct & accurate performance metrics can be achieved only by writing proper perf-tests using a benchmark framework (e.g. JMH). IMO, these kinds of naive stopwatch benchmarks are useless as they do not provide accurate, reliable metrics.

If you really want to have a stopwatch benchmark, at least measure enough iterations. Measuring 1 iteration is not accurate enough (may be 1000). Also need to measure time more precisely. Use `System.nanoTime()` instead of `System.currentTimeMillis()` (refer [this SO answer](https://stackoverflow.com/a/1776053)).

> I dont see any harm running a simple perf test locally after running the "actual" unit tests for a recordstore

The "harm" of having these types of unnecessary code segments, 
- clutters the actual test code, thus reduces the code readability and maintainability 
- vandalizes log output
- if this is not an "actual" test, then increases the build time

Reply:
> Can you explain why would you want to run a pref-test locally?

Again, this is just to get a very basic number, so that we can compare this number roughly with other records stores. I agree that, if we are going to do a perfect perf-test, we shouldn't do it even at a modular level. 

> Whose going to read these logs and compare with previous ones in the build process? :)

This is us or whoever want a rough idea about TPS numbers :). This is even helpful for others who will be writing a new recordstore for a different datasource as we are running the same test cases against it.

> Correct & accurate performance metrics can be achieved only by writing proper perf-tests using a benchmark framework (e.g. JMH). IMO, these kinds of naive stopwatch benchmarks are useless as they do not provide accurate, reliable metrics (see this OTN article)

As I said earlier this is not a perfect  perf test. We are simply calculating the time taken to persist or index a given number of records. We will not run the tests in multiple iterations as it might take a long time to complete. That's why we are running only one iteration. We shouldn't use these numbers to benchmark the DAL. These are to calculate the local perf numbers that might be helpful for the developer.

> If you really want to have a stopwatch benchmark, at least measure enough iterations. Measuring 1 iteration is not accurate enough (may be 1000). Also need to measure time more precisely. Use System.nanoTime() instead of System.currentTimeMillis() (refer this SO answer).

True, if we are running "actual" perf tests. 
> clutters the actual test code, thus reduces the code readability and maintainability

This is separate method from the rest of the code, which does not test any functionality, only a very simple perf-test. How does that affect the readability and maintainability? The method itself explains what it does and the proper log messages are used.

> vandalizes log output
> if this is not an "actual" test, then increases the build time

There is only one iteration in each method. If it really takes the build time, what we should do is, let that code be where it is, and just disable the log in the method level.

Reply:
>Again, this is just to get a very basic number, so that we can compare this number roughly with other records stores

Seems like you are treating this as a perf test even though you don't call it a perf-test. My point is, that this stopwatch perf-test is so inaccurate that we shouldn't use it even for a basic comparison.  Its like measuring the thickness of a tissue paper with a meter ruler. You cannot measure the thickness accurately & the value you get from the ruler is so off that it shouldn't be used.

>This is separate method from the rest of the code, which does not test any functionality, only a very simple perf-test. How does that affect the readability and maintainability? The method itself explains what it does and the proper log messages are used.

Since we have a fresh memory of this, we don't see this as clutter. As the time flies and our memory fades, this code will be perceived as clutter. There is a saying that every line that is unnecessary, will be a maintain hazard (because future developers won't know that that line is unnecessary & they will treat it as real code).

>There is only one iteration in each method. If it really takes the build time, what we should do is, let that code be where it is, and just disable the log in the method level.

+1 for changing the log level to debug.